### PR TITLE
HHH-15491 - Remove support for database versions that are unsupported by vendors 6.3 edition

### PR DESCRIPTION
**Pull request by jrenaat**
https://hibernate.atlassian.net/browse/HHH-15177 - Remove support for PostgreSQL versions older than 11
https://hibernate.atlassian.net/browse/HHH-15194 - Remove support for PostgreSQL Advanced Server versions older than 11
https://hibernate.atlassian.net/browse/HHH-15535 - Remove support for H2 versions older than 2.1
https://hibernate.atlassian.net/browse/HHH-15203 - Remove support for Derby versions older than 10.15

### CODE REVIEW

File: edb/edb11.Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM quay.io/enterprisedb/edb-postgres-advanced:10.22
+FROM quay.io/enterprisedb/edb-postgres-advanced:11.20
```
Comment (beikov) [inline edb/edb11.Dockerfile:1]: See the commented out block below. It's be better to use `11.20-3.3-postgis`

File: nightly.Jenkinsfile [hunk 2]
```diff
@@ -158,22 +158,22 @@ stage('Build') {
 									sh "./docker_db.sh postgresql"
 									state[buildEnv.tag]['containerName'] = "postgres"
 									break;
-								case "postgresql_10":
+								case "postgresql_11":
 									// use the postgis image to enable the PGSQL GIS (spatial) extension
 									docker.withRegistry('https://index.docker.io/v1/', 'hibernateci.hub.docker.com') {
-										docker.image('postgis/postgis:10-2.5').pull()
+										docker.image('postgis/postgis:11-2.5').pull()
```
Comment (beikov) [inline nightly.Jenkinsfile:164]: It's better to update to 11-3.3.

File: Jenkinsfile [hunk 3]
```diff
@@ -174,22 +174,14 @@ stage('Build') {
...
-									break;
 								case "edb":
 									docker.image('quay.io/enterprisedb/edb-postgres-advanced:15.2-3.3-postgis').pull()
 									sh "./docker_db.sh edb"
 									state[buildEnv.tag]['containerName'] = "edb"
 									break;
-								case "edb_10":
-									docker.image('quay.io/enterprisedb/edb-postgres-advanced:10.22').pull()
-									sh "./docker_db.sh edb_10"
+								case "edb_11":
```
Comment (beikov) [inline Jenkinsfile:182]: Remove this entry from this file please.

File: edb/edb11.Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM quay.io/enterprisedb/edb-postgres-advanced:10.22
+FROM quay.io/enterprisedb/edb-postgres-advanced:11.20-3.3-postgis
 USER root
 RUN chown -R postgres:postgres /var/lib/edb && chmod 777 /var/lib/edb && mkdir /docker-entrypoint-initdb.d
```
Comment (beikov) [inline edb/edb11.Dockerfile:3]: I think you'll need this.

```suggestion
RUN chown -R postgres:postgres /var/lib/edb && chmod 777 /var/lib/edb && rm /docker-entrypoint-initdb.d/10_postgis.sh
```

See the commented out lines below and compare that with the edb14.Dockerfile.

File: Jenkinsfile (hunk 3 above)
Comment (jrenaat) [reply Jenkinsfile:182]: I'm not sure I understand? remove the edb_11?

**Review by beikov — COMMENTED** [review]
Please also make sure that the things you removed from `H2Dialect` and `DerbyDialect` (i.e. version checks for features) are part of the `H2LegacyDialect` and `DerbyLegacyDialect` respectively.

**Comment by jrenaat** [comment]
> Please also make sure that the things you removed from `H2Dialect` and `DerbyDialect` (i.e. version checks for features) are part of the `H2LegacyDialect` and `DerbyLegacyDialect` respectively.

I did, but just to be on the safe side, I'll double check

**Comment by jrenaat** [comment]
> Please remove this `tq` file that is part of this PR

Done. No idea where this came from, prob. typing some command when the focus was where it shouldn't be ... sorry for that.

**Pull request merged**